Documentation/test-targets-usage.md

```diff
@@ -0,0 +1,108 @@
+```
+
+#### Debug tests in Visual Studio
+
+1.  Open the project's solution in Visual Studio.
+2.  Set the test project as the Startup Project.
+3.  Build (or rebuild) the solution.
+4.  Press F5 to debug the project.
+
+You can modify the XUnit command which will be executed when debugging by editing the *Command line arguments* in the *Debug* section of the test project's properties, for example to debug just a single test method.  Note that these changes will be overwritten when rebuilding the project.
+
+#### Run tests for a specified TFM and/or RID
+
+You can specify the target framework moniker (TFM) on which to run tests by using the `TestTFM` property.  Similarly, the runtime ID (RID) can be specified with the `TestNugetRuntimeId` property.
+
+For example, the first command below runs tests against .NET Framework Desktop, and the second runs tests against .NET Core 5.0 for Windows 10 x64 (using the UWP runner).
+```
+msbuild /t:Test /p:TestTFM=net46 /p:TargetGroup=netstandard1.3 /p:OSGroup=Windows_NT System.Collections.Concurrent.Tests.csproj
+msbuild /t:Test /p:TestTFM=netcore50 /p:TargetGroup=netstandard1.3 /p:OSGroup=Windows_NT /p:TestNugetRuntimeId=win10-x64 System.Collections.Concurrent.Tests.csproj
+```
+
+As the above commands suggest, it is often necessary to specify the `TargetGroup` and/or `OSGroup` together with the `TestTFM`.  For this purpose, CoreFX provides a .builds file for each test project which specifies supported configurations for the project.  These can be used to build a specified TFM by using just `FilterToTestTFM` property.  See [CoreFX's developer guide](https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/developer-guide.md#running-tests-in-a-different-tfm) for more information on this approach.
+
+#### Build and run tests with .NET Native (Windows only)
+
+Tests can be compiled and run with .NET Native by specifying the `UseDotNetNativeToolchain` property.
+
+```
+msbuild /t:BuildAndTest /p:TestTFM=netcore50aot /p:TestNugetRuntimeId=win10-x64-aot /p:UseDotNetNativeToolchain=true Microsoft.CSharp.Tests.csproj
+```
+
+#### Build and run tests against packages
+
+By default, the test project's direct dependencies (i.e. assemblies for the projects to which the test project has a ProjectReference, such as the corresponding product project) are copied from each referenced project's `$(TargetPath)`.  The `BuildTestsAgainstPackages` property provides the option to instead copy these references from a packages directory.
+```
+msbuild /t:BuildAndTest /p:BuildTestsAgainstPackages=true /p:"PackagesDrops=C:\Repos\corefx\bin\packages\Debug" System.Collections.Immutable.Tests.csproj
+```
+
+[Building Tests Against Packages](https://github.com/dotnet/corefx/blob/master/Documentation/building/build-tests-against-packages.md) has more information about doing so in CoreFX.
```

Comment:
@karajas @chcosta Is there a better way to describe the BuildTestsAgainstPackages scenario?  Is this missing anything?